test/TensorFlowRuntime/tensor.swift

```diff
@@ -173,6 +242,23 @@ TensorTests.test("WholeTensorSlicing") {
               slice2.array)
 }
 
+TensorTests.testAllBackends("AdvancedIndexing") {
```

Comment:
I can't find any tests that use strides in subscripts. Could you make sure every feature is reasonably tested for correctness?

Reply:
We have no syntax for strides currently and so I thought of strided slices as a future feature. This PR adds support in the striding mechanism and that's all tested in TensorFlow core (we're calling the TF op directly with these strides without doing any processing on the Swift side). So, I believe these tests would better be added once we add better syntax for strided slices. Does that sound reasonable?

Reply:
Since a lot of logic is being wired up in pure Swift, I wanted to be careful landing unused core features without tests. It is not entirely the case that we can't use them in subscripts. You can define static computed properties `.newAxis` and `.ellipsis` in a `TensorSliceIndexProtocol` extension. Then you should be able to do `x[1, 2...3, .newAxis, 4]`

Reply:
Right ok sounds good. I'll add these properties and associated tests later today.